packages/components/src/mobile/link-settings/link-settings-screen.native.js

```diff
@@ -20,17 +20,21 @@ const LinkSettingsScreen = ( props ) => {
 	const { inputValue = url } = route.params || {};
 
 	const onLinkCellPressed = () => {
-		navigation.navigate( 'linkPicker', { inputValue } );
+		if ( props.onLinkCellPressed ) {
+			props.onLinkCellPressed( { navigation } );
+		} else {
+			navigation.navigate( 'linkPicker', { inputValue } );
+		}
 	};
 
 	return useMemo( () => {
 		return (
 			<LinkSettings
+				{ ...props }
 				onLinkCellPressed={
 					props.hasPicker ? onLinkCellPressed : undefined
 				}
 				urlValue={ inputValue }
-				{ ...props }
```

Comment:
Prop spread relocated to prevent overriding the `onLinkCellPressed` value set here.

Reply:
> My understanding is that the `onLinkCellPressed` callback can still be overriden when passed as a prop, but via invocation within the local `onLinkCellPressed` defined above. Is that correct?

Yes, that is correct. 

> Not suggesting a change here, but curious about the scenario where a consumer might clobber this. Did you run into something like that?

If I understand your question correctly, a consumer will always be able to unset (clobber?) `onLinkCellPressed`.

```jsx
<LinkSettingsScreen onLinkCellPressed={false} />

// ...would result in the follow logic in this component...
// onLinkCellPressed={ props.hasPicker ? false : undefined }
```

A consumer will _not_ be able to enforce an `onLinkCellPressed` callback to invoke without setting `hasPicker` to a truthy value. 

```jsx
<>
  <LinkSettingsScreen onLinkCellPressed={() => console.log('> Picker!')} hasPicker={true} />
  <LinkSettingsScreen onLinkCellPressed={() => console.log('> No Picker!')} hasPicker={false} />
</>
// ...would result in...
// > Picker!
```

From searching the project, the usage in this PR is the only context where `onLinkCellPressed` is passed into this component from a parent context. So, I feel the current `hasPicker` logic is safe. WDYT?

Reply:
Thanks for the explanation David! (btw, was using [clobber](https://en.wikipedia.org/wiki/Clobbering) a bit broadly here :smile: )

> From searching the project, the usage in this PR is the only context where onLinkCellPressed is passed into this component from a parent context. So, I feel the current hasPicker logic is safe. WDYT?

I agree.